vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -75,8 +75,17 @@ def cloud_ems_hosts_attributes
         return hosts_attributes unless clouds
 
         clouds.each do |cloud_ems|
-          connection = cloud_ems.connect
-          compute_hosts  = connection.hosts.select { |x| x.service_name == "compute" }
+          compute_hosts = nil
+          begin
+            cloud_ems.with_provider_connection do |connection|
+              compute_hosts = connection.hosts.select { |x| x.service_name == "compute" }
+            end
+          rescue StandardError
+            $log.error "[#{$!.class}: #{$!}]"
```

Comment:
@blomquisg @Fryguy oook, done, how does it look like now? I took the format from amazon